docs/guides/js-waku/use-waku-react.md

```diff
@@ -0,0 +1,356 @@
+	const { encoder  } = useCreateContentPair(contentTopic);
+
+	// Wait for the node to finish loading before sending messages
+	// (isLoading === false)
+
+	// Bind push method to a node and encoder
+	const { push } = useLightPush({ node, encoder });
+
+	// Send the message using Light Push
+	const sendMessage = async (text) => {
+		if (!push || !text) {
+			return;
+		}
+		const payload = utf8ToBytes(text);
+		await push({ payload });
+	};
+	sendMessage("Hello, World!");
+}
+```
+
+:::info
+Wait for the node to finish loading before sending messages (`isLoading` === `false`).
+:::
+
+## Receive Messages Using Filter
+
+Use the `useFilterMessages()` hook to receive messages from a `Filter` subscription:
+
+```js title="App.js"
+import {
+	useCreateLightNode,
+	useCreateContentPair,
+	useFilterMessages,
+} from "@waku/react";
+import { Protocols } from "@waku/interfaces";
+
+function App() {
+	// Create and start a Light Node and wait for remote peers
+	const { node } = useCreateLightNode({
+		options: { defaultBootstrap: true },
+		protocols: [Protocols.LightPush, Protocols.Filter],
+	});
+
+	// Choose a content topic and create a decoder
+	const contentTopic = "/waku-react-guide/1/message/utf8";
+	const { decoder } = useCreateContentPair(contentTopic);
+
+	// Receive messages from Filter subscription
+	const { error, messages, isLoading } = useFilterMessages({ node, decoder });
+	console.log(messages);
+
+	// "error" captures any error that occurs while receiving messages
+	// "messages" contains a list of messages the subscription received
+	// "isLoading" indicates whether the node is still subscribing to Filter
+	// Use these to handle states and render the UI in your application
+}
+```
+
+## Retrieve Messages Using Store
+
+Use the `useStoreMessages()` hook to retrieve messages from the `Store` protocol:
+
+```js title="App.js"
+import {
+	useCreateLightNode,
+	useCreateContentPair,
+	useStoreMessages,
+} from "@waku/react";
+import { Protocols, PageDirection } from "@waku/interfaces";
+
+function App() {
+	// Create and start a Light Node and wait for remote peers
+	const { node } = useCreateLightNode({
+		options: { defaultBootstrap: true },
+		protocols: [Protocols.LightPush, Protocols.Filter],
+	});
+
+	// Choose a content topic and create a decoder
+	const contentTopic = "/waku-react-guide/1/message/utf8";
+	const { decoder } = useCreateContentPair(contentTopic);
+
+	// Set the query options
+	const options = {
+		pageDirection: PageDirection.BACKWARD,
+		pageSize: 20,
+	};
+
+	// Query the Store peer
+	const { error, messages, isLoading } = useStoreMessages({ node, decoder, options });
+	console.log(messages);
+
+	// "error" captures any error that occurs during message retrieval
+	// "messages" contains a list of messages retrieved from the Store peer
+	// "isLoading" indicates whether the node is still retrieving messages
+	// Use these to handle states and render the UI in your application
+}
+```
+
+:::info
```

Comment:
Would be good to document somewhere the optimal usage of Waku store, as set by Lorenzo. @jm-clius can you assist?